### ISSUE TITLE
PLW0120 false positive: does not count return as a break

### ISSUE THREAD

**Issue by berzi** [astral-sh/ruff#6061]
Version: `ruff 0.0.278`

I have a piece of code functionally identical to the following:
```python
def my_func():
    retries = 0
    while retries < MAX_RETRIES:
        response = send_a_request()

        try:
            check_for_errors(response)
        except ErrorThatCanBeRetried:
            time.sleep(1)
            continue

        return response
    else:
        raise MaxRetriesExceededError()
```

The intended flow is thus:
- try sending a request
- if all goes well, its response is returned
- if an uncaught error occurs, the exception propagates like usual
- if an ErrorThatCanBeRetried occurs, then the request is retried after 1 second
- if the request is retried more than MAX_RETRIES, the loop stops due to the loop condition (so, no explicit break), and the `else` block raises a different exception

I think this is all fairly standard. I believe the rule simply doesn't count a `return` as a form of loop interruption (`break`), but it probably should.

**Comment by harupy** [comment]
Is my code above equivalent to the original code?

**Comment by charliermarsh** [comment]
I think I agree with @harupy - the `else` only triggers when you break, so it seems unnecessary here, and the rule seems right IMO.
